src/core/analysis-context.ts

```diff
@@ -484,15 +484,16 @@ export class AnalysisContext {
   /**
    * Returns true if the url given is resovable by the Analyzer's `UrlResolver`.
    */
-  canResolveUrl(url: string): boolean {
+  canResolveUrl(url: PackageRelativeUrl): boolean {
     return this.resolver.canResolve(url);
   }
 
   /**
    * Resolves a URL with this Analyzer's `UrlResolver` or returns the given
    * URL if it can not be resolved.
    */
-  resolveUrl(url: string): string {
-    return this.resolver.canResolve(url) ? this.resolver.resolve(url) : url;
+  resolveUrl(url: PackageRelativeUrl): ResolvedUrl {
+    return this.resolver.canResolve(url) ? this.resolver.resolve(url) :
+                                           (url as any as ResolvedUrl);
```

Comment:
So the `url as any` before `as ResolvedUrl` is how one de-brands the PackageRelativeUrl string so it can be re-branded as a ResolvedUrl?

Reply:
Yes, exactly.